uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/services/assignment.js

```diff
@@ -1,227 +1,221 @@
-                        id: assignmentId,
-                        extra: {
-                            case_subject: assignment.subject,
-                            case_type: assignmentType,
-                            contact_id: assignment.contacts[0].contact_id,
-                            end_date: assignment.end_date,
-                            sort_name: assignment.contacts[0].sort_name,
-                            start_date: assignment.start_date
-                        }
-                    });
-
-                }
-
-                arr = $filter('orderBy')(arr, 'subject');
-                arrSearch = $filter('orderBy')(arrSearch, 'label');
-
-                $rootScope.cache.assignment = {
-                    arr: arr,
-                    obj: obj,
-                    arrSearch: arrSearch
-                };
-            }
-        }
 
-    }]);
+            deferred.resolve(data.values);
+          }, function () {
+            deferred.reject('Unable to fetch assignment types');
+          });
+
+          return deferred.promise;
+        },
+        save: function (assignment) {
+          if (!assignment || typeof assignment !== 'object') {
+            return null;
+          }
+
+          var deferred = $q.defer();
+          var params = angular.extend({
+            sequential: 1,
+            debug: config.DEBUG
+          }, assignment);
+          var val;
+
+          Assignment.save({
+            action: 'create',
+            json: params
+          }, null, function (data) {
+            if (UtilsService.errorHandler(data, 'Unable to save an assignment', deferred)) {
+              return;
+            }
 
+            val = data.values;
+            deferred.resolve(val.length === 1 ? val[0] : null);
+          }, function () {
+            deferred.reject('Unable to save an assignment');
+          });
+
+          return deferred.promise;
+        },
+        search: function (input, excludeCaseId, includeContactIds) {
+          return AssignmentSearch.query({
+            sortName: input,
+            excludeCaseIds: excludeCaseId,
+            includeContactIds: includeContactIds.join(',')
+          }).$promise;
+        },
+        updateTab: function (count) {
+          if (document.getElementsByClassName('CRM_Case_Form_Search').length) {
+            CRM.refreshParent('.CRM_Case_Form_Search');
+            return;
+          }
+
+          if (document.getElementById('tab_case')) {
+            CRM.tabHeader.updateCount('#tab_case', CRM.tabHeader.getCount('#tab_case') + (count || 0));
+            return;
+          }
+
+          if ($location.path() === '/assignments') {
+            $state.reload(true);
+          }
+        },
+        updateCache: function (data) {
+          $log.debug('updateCache');
+          $log.debug(data);
+
+          var obj = $rootScope.cache.assignment.obj || {};
+          var arr = [];
+          var arrSearch = [];
+          var assignment;
```

Comment:
The variables which have no value when initialised, should be defined with commas.
`var assignment, assignmentId ...`